docs/guides/jupyter.rst

```diff
@@ -0,0 +1,363 @@
+How to use Jupyter notebooks in Sphinx
+======================================
+
+`Jupyter`_ notebooks are a popular tool to describe computational narratives
+that mix code, prose, images, interactive components, and more.
+Embedding them in your Sphinx project allows using these rich documents as documentation,
+which can provide a great experience for tutorials, examples, and other types of technical content.
+There are a few extensions that allow integrating Jupyter and Sphinx,
+and this document will explain how to achieve some of the most commonly requested features.
+
+.. _Jupyter: https://jupyter.org/
+
+By default, `Sphinx only supports reStructuredText source files for
+documentation <https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-source_suffix>`_.
+Luckily, there are a few extensions to also allow Jupyter notebooks as source files.
+
+.. _nb-background:
+
+Background on the existing relevant extensions
+----------------------------------------------
+
+There are two main extensions that add support Jupyter notebooks as source files in Sphinx:
+nbsphinx_ and MyST-NB_. They have similar intent and basic functionality:
+both can read notebooks in ``.ipynb`` and additional formats supported by `jupytext`_,
+and are configured in a similar way.
+However, there are some differences between them:
+
+- nsphinx uses `pandoc <https://pandoc.org/>`_ to convert the Markdown from Jupyter notebooks
+  to reStructuredText and then to docutils AST,
+  whereas MyST-NB uses `MyST-Parser`_ to directly convert the Markdown text to docutils AST.
+  Both Markdown flavors are mostly equal, but they have some differences.
+- nbsphinx executes each notebook during the parsing phase,
+  whereas MyST-NB can execute all notebooks up front
+  and cache them with `jupyter-cache <https://jupyter-cache.readthedocs.io/>`_.
+  This can result in shorter build times when when notebooks are slightly modified
+  if using MyST-NB.
+- nbsphinx provides functionality to create thumbnail galleries,
+  whereas MyST-NB does not have such functionality at the moment
+  (see :ref:`below <notebook-galleries>` for more information about galleries).
+- MyST-NB allows embedding Python objects coming from the notebook in the documentation
+  (read :ref:`their "glue" documentation <myst-nb:glue>`
+  for more information)
+  and provides more sophisticated :ref:`error reporting <myst-nb:start/error-reporting>`
+  than the one nbsphinx has.
+- The visual appearance of code cells and their outputs is slightly different:
+  nbsphinx renders the cell numbers by default,
+  whereas MyST-NB doesn't.
+
+.. _nbsphinx: https://nbsphinx.readthedocs.io/
+.. _MyST-NB: https://myst-nb.readthedocs.io/
+.. _MyST-Parser: https://myst-parser.readthedocs.io/
+.. _jupytext: https://jupytext.readthedocs.io/
+
+Using which one to use depends on your use case. As general recommendations:
+
+- If you want to use :ref:`other notebook formats <other-formats>`
```

Comment:
I'm leaning towards just not telling people what they "should" do - it feels like that might become semi-arbitrary and opens up the tutorial to debate between the two projects. It seems cleaner to just focus on the "feature matrix" and UI/UX of each tool, rather than to give opinionated recommendations. Does that make sense?

Reply:
I agree that doing specific recommendations might be contentious. However, the two projects are very similar, and I (as a user) was very confused about the differences between them, so if we don't offer at least a small amount of guidance, I think we might leave our users confused or looking for answers. I was very careful here to weigh the pros and cons, and @chrisjsewell corrected me in several places where I was wrong about MyST-NB. What do you think @ericholscher ?

Reply:
I guess I still don't understand the differences, or why they both exist. If there is a short explanation of when one would be better, we should include that. If there are basically just 2 libraries that do the same thing, we should mention that, and clarify when it's better to use one. I don't know if we can do much more than that.

Reply:
That's already done in the document (there's a background section explaining the technical differences between the two). About the "why", I don't think this is the place to clarify that -- the interested readers could head over https://github.com/spatialaudio/nbsphinx/issues/420 for some historical discussion :) But the two projects have evolved since then.